### PR TITLE
refactor: rename and optimize

### PR DESCRIPTION
**Pull request by KaoImin**
This PR does:
* rename some functions and variables
* unified code style
* handle timer error

### CODE REVIEW

File: src/overlord.rs [hunk 1]
```diff
@@ -59,9 +58,9 @@ where
         interval: u64,
         timer_config: Option<DurationConfig>,
     ) -> ConsensusResult<()> {
-        let (mut smr_provider, evt_1, evt_2) = SMRProvider::new();
+        let (mut smr_provider, evt_1, evt_2) = SMR::new();
```
Comment (rev-chaos) [inline src/overlord.rs]: Params's name should also be changed, such as smr_provider here.

File: src/smr/mod.rs [hunk 2]
```diff
@@ -20,29 +20,29 @@ use crate::{error::ConsensusError, ConsensusResult};
...
+    smr_handler:   Option<SMRHandler>,
     state_machine: StateMachine,
 }
 
-impl SMRProvider {
+impl SMR {
     pub fn new() -> (Self, Event, Event) {
         let (tx, rx) = unbounded();
-        let smr = SMR::new(tx);
+        let smr = SMRHandler::new(tx);
```
Comment (rev-chaos) [inline src/smr/mod.rs]: Params's name should be renamed.

File: src/smr/state_machine.rs [hunk 3]
```diff
@@ -364,15 +364,15 @@ impl StateMachine {
         debug!("Overlord: SMR do self check");
 
         // Whenever self proposal is empty but self lock is some, is not correct.
-        if self.proposal_hash.is_empty() && self.lock.is_some() {
+        if self.epoch_hash.is_empty() && self.lock.is_some() {
```
Comment (rev-chaos) [inline src/smr/state_machine.rs]: remove self.lock.is_some()

File: src/state/process.rs [hunk 4]
```diff
@@ -44,7 +44,7 @@ enum MsgType {
 pub struct State<T: Codec, S: Codec, F: Consensus<T, S>, C: Crypto> {
     epoch_id:             u64,
     round:                u64,
-    state_machine:        SMR,
+    state_machine:        SMRHandler,
```
Comment (rev-chaos) [inline src/state/process.rs]: rename param's name

File: src/state/tests/event_test.rs [hunk 5]
```diff
@@ -50,7 +50,7 @@ async fn handle_event_test(
     let (smr_tx, mut smr_rx) = fut_unbounded();
     let (msg_tx, msg_rx) = unbounded();
 
-    let smr = SMR::new(smr_tx);
+    let smr = SMRHandler::new(smr_tx);
```
Comment (rev-chaos) [inline src/state/tests/event_test.rs]: rename param

File: src/timer.rs [hunk 6]
```diff
@@ -10,22 +10,22 @@ use futures_timer::Delay;
...
-    sender:   UnboundedSender<SMREvent>,
-    notify:   UnboundedReceiver<SMREvent>,
-    smr:      SMR,
-    epoch_id: u64,
-    round:    u64,
+    config:        TimerConfig,
+    event:         Event,
+    sender:        UnboundedSender<SMREvent>,
+    notify:        UnboundedReceiver<SMREvent>,
+    state_machine: SMRHandler,
```
Comment (rev-chaos) [inline src/timer.rs]: rename param

File: src/overlord.rs (hunk 1 above)
Comment (KaoImin) [reply src/overlord.rs]: It is named `smr` at the next line.

File: examples/salon.rs [hunk 7]
```diff
@@ -342,5 +342,5 @@ fn hash(bytes: &Bytes) -> Bytes {
 }
 
 fn timer_config() -> Option<DurationConfig> {
-    Some(DurationConfig::new(10, 10, 10, 10, 10, 10))
+    Some(DurationConfig::new(10, 10, 10))
```
Comment (rev-chaos) [inline examples/salon.rs]: Refactor durationconfig here

File: src/state/process.rs (hunk 4 above)
Comment (KaoImin) [reply src/state/process.rs]: From state's perspective, the `SMRHandler` is the state machine.

File: src/timer.rs (hunk 6 above)
Comment (KaoImin) [reply src/timer.rs]: ditto

File: src/smr/state_machine.rs (hunk 3 above)
Comment (KaoImin) [reply src/smr/state_machine.rs]: Why remove this?

File: src/overlord.rs [hunk 8]
```diff
@@ -59,9 +58,9 @@ where
...
         let smr = smr_provider.take_smr();
```
Comment (rev-chaos) [inline src/overlord.rs]: here is smr_handler

File: src/smr/state_machine.rs (hunk 3 above)
Comment (rev-chaos) [reply src/smr/state_machine.rs]: My fault, should not remove here

**Pull request merged**